tests/CPT_test.py

```diff
@@ -0,0 +1,332 @@
+from typing import Any, Dict, List, Union
+
+import pytest
+import torch
+from datasets import load_dataset
+from torch.utils.data import Dataset
+from tqdm import tqdm
+from transformers import (
+    AutoModelForCausalLM,
+    AutoTokenizer,
+    DataCollatorForLanguageModeling,
+    Trainer,
+    TrainingArguments,
+)
+
+from peft import CPTConfig, get_peft_model
+
+
+TEMPLATE = {"input": "input: {}", "intra_seperator": " ", "output": "output: {}", "inter_seperator": "\n"}
+
+MODEL_NAME = "bigscience/bloom-1b7"
```

Comment:
We should use a much smaller model for the unit tests. E.g. would  `"hf-internal-testing/tiny-random-OPTForCausalLM"` be enough here? Ideally, we can get the time per test to < 20 sec on CPU.